### ISSUE TITLE
Dataframe visualization

### ISSUE THREAD

**Issue by mguaita** [spyder-ide/spyder#21376]
### What steps will reproduce the problem?

From the variable explorer can not open a dataframe variable

### Traceback
```python-traceback
  File "/home/adminiis/anaconda3/lib/python3.8/site-packages/spyder/plugins/variableexplorer/widgets/collectionsdelegate.py", line 199, in createEditor
    if not editor.setup_and_check(value, title=key):
  File "/home/adminiis/anaconda3/lib/python3.8/site-packages/spyder/plugins/variableexplorer/widgets/dataframeeditor.py", line 952, in setup_and_check
    self.dataModel = DataFrameModel(data, parent=self)
  File "/home/adminiis/anaconda3/lib/python3.8/site-packages/spyder/plugins/variableexplorer/widgets/dataframeeditor.py", line 139, in __init__
    self.max_min_col_update()
  File "/home/adminiis/anaconda3/lib/python3.8/site-packages/spyder/plugins/variableexplorer/widgets/dataframeeditor.py", line 248, in max_min_col_update
    for dummy, col in self.df.iteritems():
  File "/home/adminiis/anaconda3/lib/python3.8/site-packages/pandas/core/generic.py", line 5989, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'iteritems'
```

## Versions

* Spyder version: 4.2.1 
* Python version: 3.8.6
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Linux 5.15.0-84-generic

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.1 (OK)
chardet >=2.0.0                 :  5.2.0 (OK)
cloudpickle >=0.5.0             :  2.2.1 (OK)
diff_match_patch >=20181111     :  20230430 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  8.12.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  4.19.0 (OK)
keyring >=17.0.0                :  24.2.0 (OK)
nbconvert >=4.0                 :  7.8.0 (OK)
numpydoc >=0.6.0                :  1.5.0 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.9.5 (OK)
pygments >=2.0                  :  2.16.1 (OK)
pylint >=1.0                    :  2.17.5 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.0             :  0.3.0 (OK)
qdarkstyle >=2.8                :  3.1 (OK)
qtawesome >=0.5.7               :  1.2.3 (OK)
qtconsole >=5.0.1               :  5.4.4 (OK)
qtpy >=1.5.0                    :  2.4.0 (OK)
rtree >=0.8.3                   :  1.0.1 (OK)
setuptools >=39.0.0             :  68.1.2 (OK)
sphinx >=0.6.6                  :  7.0.1 (OK)
spyder_kernels >=1.10.1;<1.11.0 :  1.10.1 (OK)
textdistance >=4.2.0            :  4.5.0 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3               :  3.0.0 (OK)
xdg >=0.26                      :  6.0.0 (OK)
zmq >=17                        :  25.1.1 (OK)

# Optional:
cython >=0.21                   :  3.0.2 (OK)
matplotlib >=2.0.0              :  3.5.3 (OK)
numpy >=1.7                     :  1.24.4 (OK)
pandas >=1.1.1                  :  2.0.3 (OK)
scipy >=0.17.0                  :  1.5.3 (OK)
sympy >=0.7.3                   :  1.12 (OK)
```

**Comment by ccordoba12** [comment]
Hey @mguaita, thanks for reporting. This problem is fixed in our latest version available in Anaconda (5.4.3). Please update by closing Spyder, opening a terminal and running there the commands mentioned in [our documentation](http://docs.spyder-ide.org/current/installation.html#updating-with-anaconda).

Let us know if it works for you.

**Comment by ccordoba12** [comment]
Closing due to lack of response.
